### PR TITLE
feat: Added UI for switching via dapp for custom chain id

### PR DESCRIPTION
**Pull request by NidhiKJha**
This PR is to update the UI when the dapp tries to add custom Chain Id network in MetaMask

## **Related issues**

Fixes: [https://github.com/MetaMask/MetaMask-planning/issues/2661](https://github.com/MetaMask/MetaMask-planning/issues/2661)

## **Manual testing steps**

1. Connect MetaMask to Uniswap
2. Go to Uniswap console.
3. Make an add ethereum chain call, the new popup should appear on confirmations screen

### **Before**

NA
### **After**

https://github.com/user-attachments/assets/69fa8baa-9344-4f1f-9f6e-58dee1a7c073

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by bergeron — DISMISSED** [review]

**Comment by bergeron** [comment]
Hmmmm I got this error.

I didn't built with any env vars, just yarn; yarn webpack


https://github.com/user-attachments/assets/44d98d9a-5cb5-48df-8b1a-77728315ba84

**Comment by gambinish** [comment]
> Hmmmm I got this error.
> 
> I didn't built with any env vars, just yarn; yarn webpack
> 
>  Screen.Recording.2024-09-04.at.10.22.10.AM.mov

I got this same error, but after setting `CHAIN_PERMISSIONS` env var to `true` in `.metamastkrc` I saw the expected popup behavior as expected.

Should we handle the case where this flag is not set, rather than dumping a generic error?

File: ui/pages/confirmations/confirmation/templates/add-ethereum-chain.js [hunk 1]
```diff
@@ -249,6 +250,83 @@ function getValues(pendingApproval, t, actions, history, data) {
           },
         ],
       },
+      {
+        element: process.env.CHAIN_PERMISSIONS && 'BannerAlert',
```
Comment (gambinish) [inline ui/pages/confirmations/confirmation/templates/add-ethereum-chain.js:254]: What is the expected behavior if this feature flag is not set?

**Comment by jonybur** [comment]
Similarly to @gambinish, I get an error when not using the flag, but when the flag is on, everything works well.
Screenshot 2024-09-05 at 14 00 56
Screenshot 2024-09-05 at 14 02 42
If this is desired then we can move forward

**Comment by NidhiKJha** [comment]
Nice catch @bergeron . @jonybur @gambinish without feature flag it should not crash. Here is the expected behaviour in both the cases

### With feature flag

https://github.com/user-attachments/assets/7f91ca75-8a8f-4b08-8377-399dba6b8dcc

### Without feature flag

https://github.com/user-attachments/assets/106d307b-aae8-48b6-ac3d-54c14d932c96

**Review by jonybur — APPROVED** [review]

**Review by salimtb — APPROVED** [review]

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**